tensorflow/python/estimator/inputs/queues/feeding_functions.py

```diff
@@ -169,10 +169,18 @@ def __call__(self):
         current_epoch=self._epoch,
         total_epochs=self._num_epochs)
 
+    # Get sorted indices and remember how to reorder output.
+    # This allows for the use of h5py databases which require ordered indexing.
+    sorted_integer_indexes, reorder_indexes = zip(
+        *sorted([(value, index)
+                 for index, value in enumerate(integer_indexes)]))
+    sorted_integer_indexes = list(sorted_integer_indexes)
+    reorder_indexes = list(reorder_indexes)
+
     self._trav = (integer_indexes[-1] + 1) % self._max
     feed_dict = {self._index_placeholder: integer_indexes}
     cols = [
-        column[integer_indexes]
+        column[sorted_integer_indexes][reorder_indexes]
```

Comment:
After digging into details, I think this line has a bug. You meant to put the elements into position reorder_indexes, but instead, here it is getting the elements from position reorder_indexes. I should catch this earlier. Sorry.

(I was also worrying about performance if this line does the expected assigning). 

One mini example could be:

>>> a = np.array([10, 20, 30, 40])
>>> i = [0, 1, 2, 3, 0, 1, 2, 3]                                                                                                                                                                                                                                                                                              
>>> a[i]
array([10, 20, 30, 40, 10, 20, 30, 40])
>>> sorted_integer_indexes, reorder_indexes = zip(*sorted([(value, index) for index, value in enumerate(i)]))
>>> sorted_integer_indexes = list(sorted_integer_indexes)
>>> reorder_indexes = list(reorder_indexes)
>>> reorder_indexes
[0, 4, 1, 5, 2, 6, 3, 7]
>>> a[sorted_integer_indexes][reorder_indexes]
array([10, 30, 10, 30, 20, 40, 20, 40])